app/services/fact_importer.rb

```diff
@@ -86,8 +85,13 @@ def add_new_facts
     logger.debug("Merging facts for '#{host}': added #{@counters[:added]} facts")
   end
 
+  def preload_fact_names
+    # :type is needed because custom facts usually inherits from FactName so they would be included in the list
+    fact_name_class.where(:type => fact_name_class).group(:name).maximum(:id)
```

Comment:
Thanks, it was a leftover from the code I restored, probably from when fact names were not unique. I've changed it to pluck just names/IDs.